client/state/data-layer/wpcom-api-middleware.js

```diff
@@ -90,24 +88,11 @@ export const middleware = handlers => store => next => {
 			}
 		}
 
-		// as we transition to making next() implicit we want
-		// to limit the extent of our changes so make this new
-		// function which gives us the ability to incrementally
-		// remove the uses of `next( action )` inside the handlers
-		//
-		// this guarantees that we don't double-dispatch
-		const nextActions = new Set();
-		const safeNext = a => nextActions.add( a );
-
-		handlerChain.forEach( handler => handler( store, action, safeNext ) );
+		handlerChain.forEach( handler => handler( store, action ) );
 
-		// make sure we pass along this action
-		// eventually this will return to the
-		// simpler `return next( action )`
 		if ( shouldNext( action ) ) {
```

Comment:
The purpose is to prevent HTTP responses from hitting the reducers, in turn double-dispatching actions like `POSTS_REQUEST`. That is, once we get an HTTP response back it should hit the handlers but not the reducers.